packages/react/src/createReactQueryHooks.tsx

```diff
@@ -352,15 +397,28 @@ export function createReactQueryHooks<
       prefetchInfiniteQuery(pathAndInput as any, opts as any);
     }
 
+    /**
+     * Hack to make sure errors return `status`='error` when doing SSR
+     * @link https://github.com/trpc/trpc/pull/1645
+     */
+    const actualOpts =
+      ssrEnabled && !isMounted
+        ? {
+            ...opts,
+            retry: false,
+            retryOnMount: false,
+          }
+        : opts;
```

Comment:
CC @TkDodo here's the new hack to fix this. 

Until the app is mounted on the client, if SSR is enabled, I `retry`/`retryOnMount`

Reply:
as I commented here:
- https://github.com/tannerlinsley/react-query/pull/3411#discussion_r829007744

turning off `retryOnMount` should be enough

Reply:
✅ 

If you have time and you understand the context, please skim through my changes here :)